index.js

```diff
@@ -4,56 +4,54 @@ const path = require('path');
 const crypto = require('crypto');
 const gutil = require('gulp-util');
 const through = require('through2');
+const pify = require('pify');
+
+const readFile = pify(fs.readFile);
+const stat = pify(fs.stat);
 
 // Ignore missing file error
 function fsOperationFailed(stream, sourceFile, err) {
-	if (err) {
-		if (err.code !== 'ENOENT') {
-			stream.emit('error', new gutil.PluginError('gulp-changed', err, {
-				fileName: sourceFile.path
-			}));
-		}
-
-		stream.push(sourceFile);
+	if (err.code !== 'ENOENT') {
+		stream.emit('error', new gutil.PluginError('gulp-changed', err, {
+			fileName: sourceFile.path
+		}));
 	}
 
-	return err;
+	stream.push(sourceFile);
 }
 
 const sha1 = buf => crypto.createHash('sha1').update(buf).digest('hex');
 
 // Only push through files changed more recently than the destination files
 function compareLastModifiedTime(stream, cb, sourceFile, targetPath) {
-	fs.stat(targetPath, (err, targetStat) => {
-		if (!fsOperationFailed(stream, sourceFile, err)) {
+	stat(targetPath)
+		.then(targetStat => {
 			if (sourceFile.stat && sourceFile.stat.mtime > targetStat.mtime) {
 				stream.push(sourceFile);
 			}
-		}
-
-		cb();
-	});
+		})
+		.catch(err => fsOperationFailed(stream, sourceFile, err))
+		.then(() => cb());
```

Comment:
The point of promisifying is so that we can drop the `cb()` here and in the function signature and just return a promise from the function.